runtime/src/main/java/org/corfudb/protocols/wireprotocol/TokenRequest.java

```diff
@@ -65,7 +66,7 @@ public TokenRequest(Long numTokens, Set<UUID> streams,
      * @param numTokens number of tokens to request
      * @param streams streams UUIDs required in the token request
      */
-    public TokenRequest(Long numTokens, Set<UUID> streams) {
+    public TokenRequest(Long numTokens, List<UUID> streams) {
         if (numTokens == 0) {
             this.reqType = TK_QUERY;
         } else if (streams == null || streams.size() == 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)